lib/ddtrace/buffer.rb

```diff
@@ -72,48 +126,103 @@ def close
         @closed = true
       end
     end
+  end
+
+  # Trace buffer that stores application traces and
+  # can be safely used concurrently with CRuby.
+  #
+  # Under extreme concurrency scenarios, this class can exceed
+  # its +max_size+ by up to 4%.
+  #
+  # Because singular +Array+ operations are thread-safe in CRuby,
+  # we can implement the trace buffer without an explicit lock,
+  # while making the compromise of allowing the buffer to go
+  # over its maximum limit under extreme circumstances.
+  #
+  # On the following scenario:
+  # * 4.5 million spans/second.
+  # * Pushed into a single CRubyTraceBuffer from 1000 threads.
+  # The buffer can exceed its maximum size by no more than 4%.
+  #
+  # This implementation allocates 17-93% less memory and
```

Comment:
Not a big deal but do we really want to add perf numbers in code comments? it's also a very broad range so i'm not sure how useful this comment is, maybe we can just say `'This implementation allocates significantly less memory and has modest speedup compared to Datadog::ThreadSafeBuffer'` or basically, something that won't be used against us in the feature 😅

Reply:
I was trying to ensure I had supportive arguments documented, but hard numbers are sure to be variable across any execution environment.

I want to make sure to capture information that allows for future decision making regarding trade-offs being taken here.

I'm ambivalent in keeping or removing the numbers. What do you think @brettlangdon?

Reply:
I love the transparency, but this code comment could become stale pretty quickly.

I agree with Eric, going with a simpler comment to ensure people are considerate of performance impact whenever they modify the code is 👍🏻 and then we can think about how to expose this metric in a different way.

Reply:
Thank you guys, I removed hard numbers from the comment.
There's a link to benchmarks than can be run at any time to validate if the stated performance gains still hold in this comment block already, so I'm thinking that covers the hard-numbers part.